influxql/query/compile.go

```diff
@@ -189,6 +189,7 @@ func (c *compiledStatement) compile(stmt *influxql.SelectStatement) error {
 			if err := c.subquery(source.Statement); err != nil {
 				return err
 			}
+			source.Statement.RewriteRegexConditions()
```

Comment:
Can you add a test that shows this works as expected? We should at least add a case to `TestCompile_Success` in `influxql/query/compile_test.go` to ensure this isn't breaking any compilation. Ideally we'd add a new, separate test case showing the output of compilation actually contains the expected rewrite.

Reply:
Yeah, I think it should be a seperate new test case in `compile_test.go`. I will add them.

Reply:
I add a subquery statement test case with regex condition. And thought a lot about adding a seperate test case but figure out that `query_test` is only test the final statement and compiler is working fine. It is hard to diff the ast tree outside the statement interface to make sure it works inside AST. And `RewriteRegexConditions()` is already tested by influxQL package. So the rewrite is guarantee by the package that it apply to all select statement. Any better suggestion or solution?

Reply:
I don't think I can push the change directly to your branch since it's in a forked repo; can you add the file above?

Reply:
This test file has been added. Great job !!